### ISSUE TITLE
Store photographs of the card

### ISSUE THREAD

**Issue by matthijskooijman** [brarcher/loyalty-card-locker#144]
It would make sense to, in addition to storing the card number / bar code, store photographs of the card itself. In some cases, this helps convincing store clerks that you're actually showing a customer card. It also makes displaying a card prettier (IMHO), since there's a nice image instead of just the barcode.

Additionally, the card overview could show the card front or back instead of just the card name (this is what [mobile pocket](https://play.google.com/store/apps/details?id=at.bluesource.mobilepocket&hl=en) also does. They offer pre-fab pictures for a lot of cards, but you can also take your own photographs).

This ties in with #86 (adding a logo to a card to use in the overview) and with #136 (setting a custom color for each card in the overview).

**Comment by grwlf** [comment]
+1 from here! Humans tend to memorize images much better than text, so full-size photos and card icons are must-have features.
For me, loyalty-card-locker is the only working F-Droid application for storing cards, I wish you the best.

**Comment by c-e-p-x-u-o** [comment]
I need this feature!
I have a some cards without barcode at all, and stores says: "just show the photo"!

**Comment by c-e-p-x-u-o** [comment]
What kind of help is wanted?

**Comment by brarcher** [comment]
The help would be offering an implementation with unit tests. Review https://github.com/brarcher/loyalty-card-locker/issues/86 for some implementation questions, especially regarding how logos might be imported, and how/if they would be backed up using the apps import/export feature.

**Comment by vlskrbek** [comment]
I just now looking for new application for loyalty card because:
I had CardLess+ but it stopped work (bug it shows me one card three times), also I used the Portmonka but again it hase some bugs.
So now I want to decide if try StoCard for which there are a lot of links and good refferences or use your application.
It looks very good. Here are my oppinions and plus and minus:
+ very simple
+ no adds
+ no connection to any stores so any discounts and so on it means no disturbing
+ under free license
+ I have found some support for automation
+ show diffrent type of barcodes
- cannot add any own pictureof my card for better view siearching
- cannot add own card number as a string diffrent from bar code number
- cannot display barcode in full view
- does not know searching by text

The description is that this application is doing only one thing and it try to do it in the best way. 

I am not a developer. I can help only as a beta tester. If there is somebody who can develope the new features I will be the tester.

Best Regards

**Comment by brarcher** [comment]
> It looks very good. Here are my oppinions and plus and minus:
> If there is somebody who can develope the new features I will be the tester.

Hope that the app works for you. (: 
It is a hobby for me, given free.

> I have found some support for automation

What do you mean by this? I do not think I've added features about automating its actions.

> cannot add own card number as a string diffrent from bar code number

There is a way to enter a barcode manually as text. Did you find this option when adding a new card? Is it the "Enter Card" button.

> cannot display barcode in full view

Do you mean displaying the barcode when showing the list of all the cards?

> does not know searching by text

I've not considered this. Do you mean a search feature which will look for a store in the list?

**Comment by vlskrbek** [comment]
> Hope that the app works for you. (:
> It is a hobby for me, given free.
Yes I know I do not want to push you. Take this only as my testing information.
 
> > I have found some support for automation
> What do you mean by this? I do not think I've added features about automating its actions.
I have found some discusion in Czech:
https://www.svetandroida.cz/vernostni-karty-do-mobilu/
where it is written:
---
8.4.2018 14:39
Osobně používám Loyalty Card Keychain, je malinká a rychlá, free, open source, nechce žádná zbytečná oprávnění:
https://play.google.com/store/apps/details?id=protect.card_locker
Navíc má určitou podporu pro automatizaci, kdy přes intent můžu pustit aplikaci a zobrazit konkrétní kartu. Takže mám pro často navštěvované obchody v Taskeru udělanou automatizaci, která při odemčení telefonu blízko AP u vstupu v daném obchodě spustí zobrazení “věrnostní karty” pro průchod turniketem nebo vydání čtečky, a po jejím načtení a zavření aplikace už mám rovnou připravený nákupní seznam. U pokladny opět podle stejného principu zobrazí věrnostní kartu pro naskenování.
U ostatních “peněženek” na karty jsem bohužel onu možnost spuštění s konkrétní kartou nenašel, nebo není dokumentovaná.
---
I do not know if I can belive or not but if you say that you did not do anything like that then this is just a not of that user. But There is no link or anything concrete how to do that. 
It is written in my language Czech, try to translate from Czech to English. 

> > cannot add own card number as a string diffrent from bar code number
> There is a way to enter a barcode manually as text. Did you find this option when adding a new card? Is it the "Enter Card" button.
I do not mean to enter the real barcode that is changed to picture. I mean the extra field to store another number that is diffrent from barcode. But I can put it to Notes. So this is not problem.

> > cannot display barcode in full view
> Do you mean displaying the barcode when showing the list of all the cards?
Yes I was used to tap on barcode and only the barcode was shown in ful view Nothing else. It is from Cardless+ application. But I thing this is not needed because the shops readers can easy read the standard barcode shown in detail.

> > does not know searching by text
> I've not considered this. Do you mean a search feature which will look for a store in the list?
For me this is a litle missing or I do not see it. The cards are list in alphabetical order so if I know the first letter I can find it. But If I have about 30 cards and I do not remember the name then the best way will be if it can find cards by searchin on the list. The best wy will be if after clicking on button it will wait to any keys and list only the cards that contain that char (for example B for "Bila card" but also "DM Beauty" and "Knizni kluB". If I click on second char it will show only the cards that contain the two chars "Be" so it will list only "DM BEauty".
It will be very quick filtr (search).

Best Regards

**Comment by brarcher** [comment]
> I have found some discusion in Czech:
https://www.svetandroida.cz/vernostni-karty-do-mobilu/
where it is written:

Oh, OK. I see what they are saying. There is support for opening the app to a specific card. I do recall adding that. That is what they mean by automating.

About the search, that is good feedback. My phone only had a few cards, so that is the use case I see. It is good to know that you have ~30 cards. I can see that a search would be helpful in that case. I'll consider adding that.

**Comment by vlskrbek** [comment]
Yes thats the automation. It help them to open the right card when they're in particular shop.

And thanks for considarin of filtr. You know if i have 5 cards I can leave them in pocket. But I have a little more of them :-) 
Thanks have a nice day

**Comment by Korb** [comment]
> I just now looking for new application for loyalty card because:
> I had CardLess+ but it stopped work (bug it shows me one card three times), also I used the Portmonka but again it hase some bugs.

Similar situation. Previously, I used the application "[VirtualCards-Loyalty Cards](https://play.google.com/store/apps/details?id=ro.activesoft.virtualcard)" from the company "Virtual Cards SRL" for several years, but its development has ceased, and it does not work on Google Android 12. Of all such applications on F-Droid, this seemed to me the most promising, but it was the inability to store photos of both sides of the loyalty card now does not allow me to switch to it. Of all the applications on F-Droid, this one seemed to me the most promising, but it is the lack of the ability to store photos of both sides of the loyalty card prevents me from switching to it.
